### DISCUSSION TITLE
higher L1 Sepolia Base estimation than BaseScan Sepolia estimation

### DISCUSSION THREAD

**Discussion by gitkonio** [wevm/viem#1858] [Question]
hey guys

I am trying to calculate Base Sepolia L1 fees via viem op extensions, and currently always getting higher estimation than I can see in basescan. 

Here is the example transaction: https://sepolia.basescan.org/tx/0xfa4f7fbd853ec9d06ea8d304c20aece5118c61e44e5870b6f11dcaca27967cfa
Basescan estimates 9184 L1 gas.

However, following code returns estimation of 10284 gas.

```ts
public getExtraL1Fees = async (chainId: number, hash: Hex) => {
  const isOpChain = this.isOpChain(chainId);
  const tx = await this.client.getTransaction({
    hash,
  });

  if (isOpChain) {
    const opClient = this.client.extend(publicActionsL2());

    const extraGas = await opClient.estimateL1Gas({
      account: tx.from,
      to: tx.to,
      data: tx.input,
      value: tx.value,
      nonce: tx.nonce,
      gas: tx.gas,
      maxFeePerGas: tx.maxFeePerGas,
      maxPriorityFeePerGas: tx.maxPriorityFeePerGas,
      chain: getChain(chainId),
    });

    console.log('extraGas', extraGas);
    return extraGas;
  }

  return 0n;
};
```

transaction is triggered by user-operation, so it's ERC-4337 user operation.
do you have an idea why viem estimation is quite higher than basescan estimation?

**Comment by rouzwelt** [comment]
have you tried adding the block number to your estimation call?
you need to call the gas estimation on the block that the original tx is at, just add the property: `blockNumber: tx.blockNumber`. or maybe try to work on a tx that is close to current live block number so you get a more accurate result
